Домашнее задание №7

dvdv-dev opened the pull request:
# Домашнее задание №7

Прошу @snicks92 или @crisscuola или @adamakalshchikov или @stpn111 проверить его.

Commit:
Add homework 7

Commit:
Add logger for homework 7

Добавил сам логгер

Добавил сессионную фикстуру для логгера

Изменил клиент так, чтобы он использовал логгер
* Добавил метод "__init__" с инициализацией логгера.
* Добавил вызов логгера при совершении запроса и получении ответа.
* Добавил пробел после "Host:" в реквестах, чтобы лог смотрелся более аккуратно.

Изменил фикстуру "setup" так, чтобы она использовала логгер

stpn111 reviewed (changes requested):
Привет)
Логер зачту, но по клиенту и моку есть вопрос - а где POST-запросы то?
Ну и по коду есть ряд комментариев)
На коммент про параллельность ничего делать не нужно, это просто для информации.

Ну и конфликт не забудь заресолвить.

Commit:
Fix mentioned mistakes in homework 7

Добавил в клиент метод "post_user", позволяющий совершать POST запросы с
информацией о новом пользователе.
* Тесты на добавление пользователя теперь используют данный метод,
включая новые тесты: "test_add_user_via_post" и
"test_add_user_without_first_name_via_post".
Тест "test_add_user" был переименован в "test_add_user_via_put".
Тест "test_add_user_without_first_name" был переименован в
"test_add_user_without_first_name_via_put".
Оба переименованных теста всё также используют метод "put_user".
* Данный метод косвенно использует добавленный в сервер метод
"request_post_user", позволяющий принимать POST запросы.
* В рамках данного исправления создал метод "_post_or_put_user",
который унифицирует совершение POST и PUT запросов. Метод "put_user"
теперь представляет из себя вызов данного метода, как и "post_user".
* В рамках данного исправления добавил метод "check_response" в класс
"BaseCase", который унифицирует проверки ответов на запросы. Также были
изменены все тесты так, чтобы использовать данный метод.

Удалил более не релевантные комментарии из "server.py" и "test.py".

Добавление пользователя в тестах на изменение и удаление теперь
происходит путём непосредственного изменения словаря пользователей.

Новые имя и фамилия в тесте на изменение теперь генерируются при
помощи faker.
* В тестах теперь возможно получить доступ к экземпляру faker через
переменную "self.faker_instance".

Изменён метод "request_put_user" в сервере для более корректной работы:
проверка отсутствия имени и фамилии в запросе теперь происходит после
проверки существования пользователя.

Для поддержания полного покрытия после поправок были также добавлены
тесты: "test_add_existing_user", "test_edit_user_with_empty_input".

dvdv-dev committed:
Merge branch 'main' into SDET-Python-homework-7
